bot/cogs/math.py

```diff
@@ -1,21 +1,48 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 
-
+import asyncio
+import sys
+from contextlib import suppress
 from io import BytesIO
 from re import search
-
+from subprocess import PIPE, Popen, STDOUT, TimeoutExpired  # noqa: B404
 
 from aiohttp import ClientSession
 
 from discord import File
 from discord.ext.commands import command
 
-from sympy import latex
-from sympy.parsing.sympy_parser import parse_expr
 
+LATEX_URL = "https://latex2png.com"
+
+
+async def run_sympy(sympy_code: str, calc: bool = False, timeout: int = 10) -> str:
+    if calc:
+        code_ = "parse_expr(sys.argv[1]).doit()"  # Run the expression
+    else:
+        code_ = "parse_expr(sys.argv[1], evaluate=True)"  # Just latexify it without running
+
+    if "__" in sympy_code:
+        # They're trying to exploit something, raise an error
+        raise TypeError("'__' not allowed in sympy code")
+
+    proc = Popen([sys.executable, "-c",  # noqa: B603
+                  "import sys,sympy;from sympy.parsing.sympy_parser import parse_expr;"
+                  f"print(sympy.latex({code_}))", sympy_code],
+                 stdout=PIPE, stderr=STDOUT,  # reroute all to stdout
+                 env={})  # Disable environment variables for security
```

Comment:
Could this line and the one above be indented by one more space? They're currently indented by an odd number of spaces, and aren't consistent with the lines above.